[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #4687

Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

Differential Revision: [D61200610](https://our.internmc.facebook.com/intern/diff/D61200610)

larryliu0820 committed:
[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa

Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa"


Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa"


Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa"


Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61200610).

larryliu0820 committed:
Update on "[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa"


Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

Differential Revision: [D61200610](https://our.internmc.facebook.com/intern/diff/D61200610)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61200610).

larryliu0820 committed:
Update on "[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa"


Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

Differential Revision: [D61200610](https://our.internmc.facebook.com/intern/diff/D61200610)

[ghstack-poisoned]

mcr229 commented on examples/models/llava/install_requirements.sh:
```diff
@@ -6,39 +6,7 @@
-# Newer transformer (4.38) will give TypeError: LlavaLlamaForCausalLM.forward() got an unexpected keyword argument 'cache_position'
-pip install timm==0.6.13
-pip install transformers==4.37.2
+pip install transformers
```
Should we also run 
`bash ./install_requirements` and
`bash examples/models/llama2/install_requirements.sh` as well

larryliu0820 replied:
I think user should call it outside

mcr229 replied:
I see, the ./install_requirements makes sense, but at least for the llama2, I don't think its immediately obvious that you have to install the requirements for llama2 in order for llava to work.

guangy10 replied:
@larryliu0820 great to me we are moving to the HF model. I think we should have no blocker to clean up all the ad-hoc setup requirements https://github.com/pytorch/executorch/pull/4320?

mcr229 commented on .github/workflows/pull.yml:
```diff
@@ -204,28 +204,14 @@ jobs:
         # install Llava requirements
         bash examples/models/llama2/install_requirements.sh
         bash examples/models/llava/install_requirements.sh
 
```
why don't we run export_llava.py anymore? looks like test_llava still requires the model to have already been exported?

larryliu0820 replied:
The last test in test_llava.py exports and tests llava

larryliu0820 committed:
Update on "[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa"


Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

Differential Revision: [D61200610](https://our.internmc.facebook.com/intern/diff/D61200610)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa"


Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

Differential Revision: [D61200610](https://our.internmc.facebook.com/intern/diff/D61200610)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa"


Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

Differential Revision: [D61200610](https://our.internmc.facebook.com/intern/diff/D61200610)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa"


Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

Differential Revision: [D61200610](https://our.internmc.facebook.com/intern/diff/D61200610)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61200610).

larryliu0820 committed:
Update on "[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa"


Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

Differential Revision: [D61200610](https://our.internmc.facebook.com/intern/diff/D61200610)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61200610).

larryliu0820 committed:
Update on "[llava] Use huggingface LLaVA instead of depending on third-party/LLaVa"


Currently we depend on third-party/LLaVA for llava model definition.
This is hard to use because we have to pull LLaVA in as a git submodule
and install from there. It also breaks a lot of dependency assumptions.

This PR removes `third-party/LLaVA`, in favor of huggingface llava model
definition.

Differential Revision: [D61200610](https://our.internmc.facebook.com/intern/diff/D61200610)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61200610).
